File: src/termostato/Termostato.java
package termostato;

import java.util.*;

public class Termostato {
	private static Scanner in;

	public boolean casoPrueba() {
		int n = in.nextInt();
		int e = in.nextInt();
		int a = in.nextInt();
		int actual;
		int minutos = 0;
		boolean encendido = false;

		if (n == 0 && e == 0 && a == 0) {
			return false;
		}

		for (int i = 0; i < n; i++) {
			actual = in.nextInt();

			if (encendido && actual >= a) {
				encendido = false;
			} else if (encendido || actual < e) {
				minutos++;
				encendido = true;
			}
		}

		System.out.printf("%d\n", minutos);

		return true;
	}

	public static void main(String args[]) {
		in = new Scanner(System.in);
		Termostato app = new Termostato();
		while (app.casoPrueba())
			;
	}
}

File: C++/termostato.cpp

#include <iostream>
#include "soluciones.h"

static bool casoPrueba() {
	int n;
	int min;
	int max;

	int temp;
	bool encendido = false;
	int minutos = 0;

	std::cin >> n;
	std::cin >> min;
	std::cin >> max;

	if (n == 0 && min == 0 && max == 0) {
		return false;
	}

	for (int i = 0; i < n; i++) {
		std::cin >> temp;

		if (encendido && temp >= max) {
			encendido = false;
		}
		else if (encendido || temp < min) {
			encendido = true;
			minutos++;
		}
	}

	std::cout << minutos << '\n';

	return true;
}

void lanzadores::termostato() {
	while (casoPrueba())
		;
}